superset-frontend/src/dashboard/actions/nativeFilters.ts

```diff
@@ -74,6 +74,7 @@ export const setFilterConfiguration = (
     filterConfig,
   });
   const { id, metadata } = getState().dashboardInfo;
+  const oldFilters = getState().nativeFilters?.filters;
```

Comment:
Why is this called `oldFilters`? Shouldn't this be `newFilters`?

Reply:
filters data that we had before got new data from server/FiltersConfigModal

Reply:
Ok, now I understand. With the introduction of yet a new property, the properties in `fillNativeFilters` could probably use some renaming, as there's lots of different state going through that function. Currently it's

```ts
function fillNativeFilters(
  data: FilterConfiguration,
  cleanState: DataMaskStateWithId,
  draft: DataMaskStateWithId,
  filters?: Filters,
) {
...
}
```
I think this would be clearer if it were renamed something like this:
```ts
function fillNativeFilters(
  filterConfig: FilterConfiguration,
  mergedDataMask: DataMaskStateWithId,
  draftDataMask: DataMaskStateWithId,
  currentFilters?: Filters,
) {
...
}
```
My point being, it's becoming increasingly difficult to follow the logic here as it's getting more complex. So trying to be as explicit as possible when naming variables can go a long way in making the code more readable.

FYI this is not a critique to this or other related code in general, I just want to raise the point that any effort put into code readability can go a long way helping others contribute to this code once the dust settles.